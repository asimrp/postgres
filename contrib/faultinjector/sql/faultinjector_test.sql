CREATE EXTENSION faultinjector;

-- start with a clean slate
select inject_fault('all', 'reset');

-- inject fault of type skip
select inject_fault('checkpoint', 'skip', '', '', '', 1, 2, 0);

-- wait for fault triggered 0 times, should not block
select wait_until_triggered_fault('checkpoint', 0);

-- trigger a checkpoint which will trigger the fault
checkpoint;
select wait_until_triggered_fault('checkpoint', 1);

-- check status
select inject_fault('checkpoint', 'status');
select inject_fault('checkpoint', 'reset');
