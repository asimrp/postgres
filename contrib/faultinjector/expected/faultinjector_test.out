CREATE EXTENSION faultinjector;
-- start with a clean slate
select inject_fault('all', 'reset');
 inject_fault 
--------------
 Success:
(1 row)

-- inject fault of type skip
select inject_fault('checkpoint', 'skip', '', '', 1, 2, 0);
 inject_fault 
--------------
 Success:
(1 row)

-- wait for fault triggered 0 times, should not block
select wait_until_triggered_fault('checkpoint', 0);
 wait_until_triggered_fault 
----------------------------
 Success:
(1 row)

-- trigger a checkpoint which will trigger the fault
checkpoint;
select wait_until_triggered_fault('checkpoint', 1);
 wait_until_triggered_fault 
----------------------------
 Success:
(1 row)

-- check status
select inject_fault('checkpoint', 'status');
                                                                                         inject_fault                                                                                         
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Success: fault name:'checkpoint' fault type:'skip' database name:'' table name:'' start occurrence:'1' end occurrence:'2' extra arg:'0' fault injection state:'triggered' num times hit:'1' +
 
(1 row)

select inject_fault('checkpoint', 'reset');
 inject_fault 
--------------
 Success:
(1 row)

